src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleRequestSender.cs

```diff
@@ -770,5 +778,30 @@ private void HandleCustomHostLaunch(ITestHostLauncher customHostLauncher, Messag
                 this.communicationManager.SendMessage(MessageType.CustomTestHostLaunchCallback, ackPayload, this.protocolVersion);
             }
         }
+
+        private void AttachDebuggerToProcess(ITestHostLauncher customHostLauncher, Message message)
+        {
+            var ackPayload = new ProxyAttachDebuggerToProcessAckPayload() { Attached = false, ErrorMessage = null };
+
+            try
+            {
+                var pid = this.dataSerializer.DeserializePayload<int>(message);
+                ackPayload.Attached = ((ITestHostLauncher2)customHostLauncher).AttachDebuggerToProcess(pid);
```

Comment:
>ITestHostLauncher2 [](start = 40, length = 18)

So if an IDE upgrades to the latest version of TP/Translation layer, would they need to make any changes? Implement ITestHostLauncher2 for instance to support debugging? Looks like we bail in DesignModeClient if the protocol version is lower but it isn't clear if\where we gracefully fallback. This is probably why an RFC would help.

Reply:
Yes this part is currently not covered in the RFC. We should update it.

Reply:
BTW the assumption was that the only IDE we support at the moment is VS. Do we have other IDEs that depend on *Vs*TestConsole?

Reply:
oh we do.... Rider, VS Code and perhaps even VS for Mac (there might be others). I thought we discussed this earlier. There isn't documentation of a full list though - I found [this ](https://github.com/microsoft/vstest/pull/1627#discussion_r192655976)comment that mentions two from the list above. It appears Rider operates directly on the json contract and does not use the Translation Layer APIs.

Reply:
I think we can still fallback to the old way of doing things if the IDE protocol version is less than what we support for Attach. That should hopefully not be a big change and would preserve compat.

Reply:
Thanks for catching this @AbhitejJohn! @cvpoienaru This is something that would need to be addressed (i.e. maintain old flow for IDEs that don't implement ITestHostLauncher2).

Reply:
Awesome catch @AbhitejJohn ! 👍 Fixed this by making sure we use the old `LaunchAndAttach` workflow whenever we don't have access to an `ITestHostLauncher2` implementation inside `DefaultTestHostManager.cs`/`DotnetTestHostManager.cs`. Also took care of the unconditional casts to `ITestHostLauncher2` so that we now return false if the cast fails.